compiler/noirc_evaluator/src/ssa/ir/value.rs

```diff
@@ -69,4 +191,20 @@ impl Value {
             Value::ForeignFunction { .. } => &Type::Function,
         }
     }
+
+    pub(crate) fn map_values<S>(self, f: impl Fn(ValueId<R>) -> ValueId<S>) -> Value<S> {
```

Comment:
I was a bit confused at first by this method because `f(...)` is never called... so it seems this method is just changing the type of `R` to `S`, right? I wonder if `std::mem::transmute` would be equivalent... though maybe Rust realized there's nothing to do and both are equivalent. In any case, I think this method is only used once in our code so it's fine.

Reply:
It's used [here](https://github.com/noir-lang/noir/pull/6487/files/cb121a57eeb6f902a5ec9dab8ada79beb6fc3f5f#diff-41b8b9eaa4fcb8cbbbb23ea9d7f25e2d28c81ddcd7702a4bdf86fdb27deec65eR203) :

```rust
Value::Array { array: array.into_iter().map(f).collect(), typ }
```

Thanks for the pointing out `std::mem::transmute` though, I knew I had seen this somewhere but forgot. Maybe we can use it when converting `ValueId<Resolved>` to `ValueId<Unresolved>` and perhaps `Id<Value>` to `ValueId`. Although my assumption is that it doesn't matter because these new type wrappers disappear at runtime 🤞